### PR TITLE
BREAKING--Convert datastar models to dynastar models

### PR DESCRIPTION
**Pull request by cfellin1**
This is a duplicate of https://github.com/godaddy/warehouse.ai-status-api/pull/20, but not forked, so that travis will run.

## Summary

Part of warehouse dynamo migration

## Changelog

BREAKING--Convert datastar models to dynastar models

## Test Plan

Updated unit and integration tests

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,11 +4,18 @@ language: node_js
 node_js:
   - "8"
   - "10"
+  - "12"
+before_install:
+# package-lock.json was introduced in npm@5
+- '[[ $(node -v) =~ ^v9.*$ ]] || npm install -g npm@latest' # skipped when using node 9
```
Comment (msluther) [inline .travis.yml]: Do we need these?

File: package.json [hunk 2]
```diff
@@ -8,8 +8,9 @@
   },
   "scripts": {
     "pretest": "npm run swagger",
-    "test": "nyc --reporter=lcov --reporter=text mocha test/*.test.js",
+    "test": "nyc --reporter=lcov --reporter=text mocha --require test/setup test/*.test.js --timeout 10000",
     "lint": "eslint-godaddy *.js preboots/*.js routes/*.js test/*.js",
```
Comment (msluther) [inline package.json]:
```suggestion
    "lint": "eslint-godaddy *.js \"{preboots,routes,test}/**/*.js\"",
```

File: test/status-handler.test.js [hunk 3]
```diff
@@ -183,28 +186,44 @@ describe('Status-Handler', function () {
...
+        clients: [dynamoDriver],
+        waitSeconds: 60
+      });
       await handler.models.ensure();
+      const cleanupSpec = handler._transform(fixtures.singleEvent, 'counter');
+
+      // I know this is bad code smell, but localstack keeps queueing up a
+      // mysterious insert somewhere in the first second of its launch, and
+      // this is a workaround for that.
+      const wait = ms => new Promise((r) => setTimeout(r, ms));
```
Comment (msluther) [inline test/status-handler.test.js]: nit:
```suggestion
      const wait = ms => new Promise(r => setTimeout(r, ms));
```

**Comment by jcrugzz** [comment]
@Swaagie while i agree this should be more sane, its something specific to localstack mocha behavior and doesn't impact our deployed use case. Id rather move forward since I have limited timeline and would like to get this tested in a real scenario but i dont have a problem there being other work to try and fix this.

**Comment by cfellin1** [comment]
The bad news is that the new integration test failures that started when I merged in the webhook changes appear to also only happen in combination with other test suites. We may just need to figure out how to get localstack to behave like a normal child.

**Comment by Swaagie** [comment]
I'm curious how do we know this is only related to `localstack`? To me it feels more like different write mechanics for dynamodb compared to cassandra. Some other warehouse repo's also have to wait before a cleanup can be done. In general the whole non-deterministic approach of connections to AWS.DynamoDB feels wonky to me, but that is not in our control. I'll merge so we can update our internal repo and move forward.

**Pull request merged**